command_runner/runner.go

```diff
@@ -29,14 +30,22 @@ type Runner interface {
 	RunSubCmdByName(cmdName string, subCommand string, c *cli.Context) (err error)
 }
 
+type Factory interface {
+	GetByCmdName(cmdName string) (cmd command.Command, err error)
+	GetBySubCmdName(cmdName string, subCmdName string) (cmd command.Command, err error)
+	CommandMetadatas() []command_metadata.CommandMetadata
+}
+
 type ConcreteRunner struct {
```

Comment:
Just noting in passing, there's no need to export this struct, we could write it as `concreteRunner`.